src/lib/mongoBackend/connectionOperations.cpp

```diff
@@ -304,6 +319,7 @@ bool collectionInsert
   {
     connection->insert(col.c_str(), doc);
     releaseMongoConnection(connection);
+    TIME_STAT_MONGO_WRITE_WAIT_STOP();
```

Comment:
The reason not to put the 'end' statement in 'erroneous paths' of the program is that I thought those measurements might not be very interesting. The operation failed, so the time measured could be just anything ... 
There is no problem with the logic, the STOP macro isn't used so whatever was 'recorded' by the START macro is thrown away.

I don't have any problem in measuring erroneous paths as well, I just didn't find it very interesting ...